ddtrace/tracer.py

```diff
@@ -271,18 +272,18 @@ def start_span(self, name, child_of=None, service=None, resource=None, span_type
         # add it to the current context
         context.add_span(span)
 
+        # check for new process if runtime metrics worker has already been started
+        if self._runtime_worker:
+            self._check_new_process()
+
         # update set of services handled by tracer
-        if service:
+        if service and service not in self._services:
```

Comment:
Noticed by looking at logs that we were updating `self._services` and constant tags with each span, which is definitely overkill. So added this membership test before adding a service.